devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -85,90 +79,234 @@ metadata {
 import physicalgraph.zwave.commands.doorlockv1.*
 import physicalgraph.zwave.commands.usercodev1.*
 
+/**
+ * Called on app installed
+ */
 def installed() {
 	// Device-Watch pings if no device events received for 1 hour (checkInterval)
 	sendEvent(name: "checkInterval", value: 1 * 60 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
 
+/**
+ * Executed when the user taps on the 'Done' button on the device settings screen. Sends the values to lock.
+ *
+ * @return hubAction: The commands to be executed
+ */
+def updated() {
+	if (device.rawDescription.contains("mfr:0129") && device.getType() != "Yale Z-Wave Lock") {
+		log.trace "[DTH] Setting device type to 'Yale Z-Wave Lock' for device ${device.displayName}"
+		setDeviceType("Yale Z-Wave Lock")
+		return null
+	}
+	
+	// Device-Watch pings if no device events received for 1 hour (checkInterval)
+	sendEvent(name: "checkInterval", value: 1 * 60 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	def hubAction = null
 	try {
-		if (!state.init) {
+		def cmds = []
+		if (!state.init || !state.configured) {
 			state.init = true
-			// Wait long enough for behind-the-scenes z-wave magic to finish, but be quick enough before hub goes back into inclusion and blocks us
-			response(["delay 2000"] + secureSequence([zwave.doorLockV1.doorLockOperationGet(), zwave.batteryV1.batteryGet()], 2200))
+			log.debug "Returning commands for lock operation get and battery get"
+			if (!state.configured) {
+				cmds << doConfigure()
+			}
+			cmds << refresh()
+			cmds << reloadAllCodes()
+			if (!state.MSR) {
+				cmds << zwave.manufacturerSpecificV1.manufacturerSpecificGet().format()
+			}
+			if (!state.fw) {
+				cmds << zwave.versionV1.versionGet().format()
+			}
+			hubAction = response(delayBetween(cmds, 4200))
 		}
 	} catch (e) {
-		log.warn "installed() threw $e"
+		log.warn "updated() threw $e"
 	}
+	hubAction
 }
 
-def updated() {
-	// Device-Watch pings if no device events received for 1 hour (checkInterval)
-	sendEvent(name: "checkInterval", value: 1 * 60 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+/**
+ * Configures the device to settings needed by SmarthThings at device discovery time
+ *
+ */
+def configure() {
+	log.trace "[DTH] Executing 'configure()' for device ${device.displayName}"
+	def cmds = doConfigure()
+	log.debug "Configure returning with commands := $cmds"
+	cmds
 }
 
+/**
+ * Returns the list of commands to be executed when the device is being configured/paired
+ *
+ */
+def doConfigure() {
+	log.trace "[DTH] Executing 'doConfigure()' for device ${device.displayName}"
+	state.configured = true
+	if (device.rawDescription.contains("mfr:003B")) {
```

Comment:
A better way to do this is `zwaveInfo.mfr == "003B"`

Reply:
I'll make this change in the next commit. We are going to submit more changes to the DTH.